File: Data Structures & Algorithms/minimum-difference-between-highest-and-lowest-of-k-scores/submission-1.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:

        nums.sort()
        left = 0
        ans = float('inf')
        for right in range(k - 1, len(nums)):

            if right - left + 1 > k:
                left += 1
            ans = min(ans, nums[right] - nums[left])
        
        return ans 
        

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> seen = new HashSet<>();
        int ans = 0;
        int left = 0;
        for(int right = 0; right < s.length(); right++) {
            while (seen.contains(s.charAt(right))) {
                seen.remove(s.charAt(left++));
            }
            seen.add(s.charAt(right));
            ans = Math.max(ans, right - left + 1);
        }
        return ans;
    }
}
